pkg/test/kube/accessor.go

```diff
@@ -244,6 +244,41 @@ func (a *Accessor) WaitUntilDaemonSetIsReady(ns string, name string, opts ...ret
 	return err
 }
 
+// WaitUntilServiceIsReady will wait until the service with the given name/namespace is present, and have at least
+// one usable endpoint.
+func (a *Accessor) WaitUntilServiceIsReady(ns string, name string, opts ...retry.Option) (*kubeApiCore.Service, error) {
+	var service *kubeApiCore.Service
+	err := retry.UntilSuccess(func() error {
+
+		s, err := a.GetService(ns, name)
+		if err != nil {
+			return err
+		}
+
+		endpoints, err := a.GetEndpoints(ns, name, kubeApiMeta.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if len(endpoints.Subsets) == 0 {
+			return fmt.Errorf("%s/%v endpoint not ready: no subsets", ns, name)
+		}
+
+		for _, subset := range endpoints.Subsets {
+			if len(subset.Addresses) > 0 {
```

Comment:
There are some use cases where there will be no endpoints. It might be better to check that the list of notReadyAddresses is empty.

Reply:
I thought about it after chatting. I think we want to preserve this behavior, for this particular use-case. This method gets called right after deploying, trying to ensure that there is at least one available endpoint. Looking for not-ready subsets can potentially cause problems, as without a deployment that is trying to start-up (i.e. due to timings) this set can come back as empty, even though we're very early in the startup stage.

The external service case should be covered by a different method. I renamed this one to make it explicitly clear what it does.

Reply:
ok, fair enough ... well, can you do both? make sure that `len(subset.Addresses) > 0 && len(subset.NotReadyAddresses) == 0`?